src/target/arc_ocd.c

```diff
@@ -141,6 +141,13 @@ int arc_ocd_deassert_reset(struct target *target)
 	/* deassert reset lines */
 	jtag_add_reset(0, 0);
 
+	if (!target->reset_halt && target->state == TARGET_RESET)
+	{
```

Comment:
Also please add `LOG_DEBUG` message here that "starting cpu execution after reset" or something like that.